test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpParentTest.java

```diff
@@ -0,0 +1,159 @@
+                {"//Customer[1]/namespace::*/..", "//Customers"},
+                {"/Customers/comment()/..", "//Customers"},
+                {"//*[parent::Customers][1]", "//Customer"},
+                {"//*[parent::Customers and @id='x1']", "//Customer"},
+
+                // document root
+                {"/Customers/parent::node()", "/"},
+                {"/Customers/..", "/"},
+        };
+    }
+
+    /*
+     * DataProvider for XPath expressions which should provide no parent node.
+     * Data columns:
+     *  see parameters of the test "testZeroParentNodeExp"
+     */
+    @DataProvider(name = "noParentNode")
+    public Object[][] getZeroParentNodeExp() {
+        return new Object[][]{
+                // parent::* never return document root
+                {"/Customers/parent::*"},
+
+                // parent of document root
+                {"/.."},
+        };
+    }
+
+    /*
+     * DataProvider for XPath relative expressions which should provide a parent node.
+     * Data columns:
+     *  see parameters of the test "testRelativeParentExp"
+     */
+    @DataProvider(name = "relativeParent")
+    public Object[][] getRelativeParentExp() {
+        return new Object[][]{
+                {"/Customers", "comment()/parent::*"},
+                {"/Customers/Customer[1]", "Name/parent::*"},
+                {"/Customers/Customer[1]", "text()/parent::*"},
+                {"/Customers/Customer[1]", "@id/parent::*"},
+                {"/Customers", "namespace::*/parent::*"},
+                {"/Customers", "comment()/parent::*"},
+                {"/Customers/Customer[1]", "../Customer[1]/Name/.."},
+                {"/Customers/Customer[1]", "../Customer[1]/text()/.."},
+                {"/Customers/Customer[1]", "../Customer[1]/@id/.."},
+                {"/Customers", "namespace::*/.."},
+        };
+    }
+
+    /**
+     * Verifies that XPath expressions provide one and only parent node.
+     *
+     * @param exp       XPath expression
+     * @param parentExp XPath parent node expression
+     * @throws Exception if test failed
+     */
+    @Test(dataProvider = "oneParentNode")
+    void testOneParentNodeExp(String exp, String parentExp)
+            throws Exception {
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        Node expected = xPath.evaluateExpression(parentExp, doc, Node.class);
+
+        testExp(doc, exp, expected, Node.class);
+
+        XPathNodes nodes = xPath.evaluateExpression(exp, doc, XPathNodes.class);
+
+        Assert.assertEquals(nodes.size(), 1);
+    }
+
+    /**
+     * Verifies that XPath expressions provide no parent node.
+     *
+     * @param exp XPath expression
+     * @throws Exception if test failed
+     */
+    @Test(dataProvider = "noParentNode")
+    void testZeroParentNodeExp(String exp)
+            throws Exception {
+        testExp(doc, exp, null, Node.class);
+    }
+
+    /**
+     * Verifies that XPath relative expressions provide a parent node.
```

Comment:
The context node happens to be the parent node of the 2nd statement. Needs to document it in the description and the params below.